bazel/envoy_build_system.bzl

```diff
@@ -0,0 +1,41 @@
+# References to Envoy external dependencies should be wrapped with this function.
+def envoy_external_dep_path(dep):
+    return "//external:%s" % dep
+
+ENVOY_COPTS = [
+    "-ggdb3", "-fno-omit-frame-pointer", "-Wall", "-Wextra", "-Werror",
```

Comment:
I'm wondering where this would go in the repository. Let's say we do want our own standard set of additions for fastbuild/opt/debug, such as -ggdb3. Would we put this in $(ENVOY_SRC)/.bazelrc? If we put it here, my reading of the docs is that ~/.blazerc is then ignored. This would be bad as we disallow user configurability.